beacon-chain/light-client-network.md

```diff
@@ -0,0 +1,179 @@
+
+### Wire Protocol
+
+The [Portal wire protocol](./portal-wire-protocol.md) is used as the wire protocol for the Beacon Chain Light Client network.
+
+#### Protocol Identifier
+
+As specified in the [Protocol identifiers](./portal-wire-protocol.md#protocol-identifiers) section of the Portal wire protocol, the `protocol` field in the `TALKREQ` message **MUST** contain the value of `0x501A`.
+
+#### Supported Messages Types
+
+The beacon chain light client network supports the following protocol messages:
+
+- `Ping` - `Pong`
+- `Find Nodes` - `Nodes`
+- `Find Content` - `Found Content`
+- `Offer` - `Accept`
+
+#### `Ping.custom_data` & `Pong.custom_data`
+
+In the beacon chain light client network the `custom_payload` field of the `Ping` and `Pong` messages is the serialization of an SSZ Container specified as `custom_data`:
+
+```
+custom_data = Container(data_radius: uint256)
+custom_payload = serialize(custom_data)
+```
+
+### Routing Table
+
+The Beacon CHain Light Client Network uses the standard routing table structure from the Portal Wire Protocol.
+
+### Node State
+
+#### Data Radius
+
+The Beacon Chain Light Client Network includes one additional piece of node state that should be tracked. Nodes must track the `data_radius`
+from the Ping and Pong messages for other nodes in the network. This value is a 256 bit integer and represents the data that
+a node is "interested" in.
+
+We define the following function to determine whether node in the network should be interested in a piece of content:
+
+```
+interested(node, content) = distance(node.id, content.id) <= node.radius
+```
+
+A node is expected to maintain `radius` information for each node in its local node table. A node's `radius` value may fluctuate as the contents of its local key-value store change.
+
+A node should track their own radius value and provide this value in all Ping or Pong messages it sends to other nodes.
+
+### Data Types
+
+The beacon chain light client DHT stores the following data items:
+
+* LightClientBootstrap
+* LightClientUpdate
+
+The following data objects are ephemeral and we store only the latest values:
+
+* LightClientFinalityUpdate
+* LightClientOptimisticUpdate
+
+#### Constants
+
+We define the following constants which are used in the various data type definitions:
+
+```py
+# Maximum number of `LightClientUpdate` instances in a single request
+MAX_REQUEST_LIGHT_CLIENT_UPDATES = 2**7  # = 128
+```
+
+#### LightClientBootstrap
+
+```
+light_client_bootstrap_key = Container(block_hash: Bytes32)
+selector                   = 0x00
+
+content                    = SSZ.serialize(LightlientBootstrap)
+content_key                = selector + SSZ.serialize(light_client_bootstrap_key)
+```
+
+#### LightClientUpdatesByRange
+
+```
+light_client_update_keys   = Container(start_period: uint64, count: uint64)
+selector                   = 0x01
+
+content                    = SSZList(LightClientUpdate, max_lenght=MAX_REQUEST_LIGHT_CLIENT_UPDATES)
```

Comment:
Needs to be fork-aware, the individual items may have different types. 
See https://github.com/ethereum/beacon-APIs/pull/247

(Same for the other contents, but there it can be solved by simply prefixing the content with the corresponding forkDigest)

Reply:
> (Same for the other contents, but there it can be solved by simply prefixing the content with the corresponding forkDigest)

Could you elaborate on this? My understanding is that we need to return `ForkDigest` only for every `LightClientUpdate` payload, what other contents do you mean?

Reply:
`LightClientFinalityUpdate` / `LightClientOptimisticUpdate` / `LightClientBootstrap` should all be fork aware for future proofing.

Concretely, starting with Capella, those structures may be modified to also include the EL block hash. So, there needs to be some indicator for the receiver to know what structure to parse. Whether that is the fork-digest like on libp2p/REST, or just a simple enum with 1=altair, 2=bellatrix, 3=capella or something else doesn't matter. Technically, it is also possible to rely that `attested_header.slot` is kept at offset 0, but rather keep that flexible; the 1-4 extra bytes don't hurt.